src/lsm/grid.zig

```diff
@@ -351,19 +327,16 @@ pub fn GridType(comptime Storage: type) type {
             );
             util.copy_disjoint(.exact, u8, cached_block, completed_write.block);
 
-            grid.write_iops.release(iop);
-
             // Start a queued write if possible *before* calling the completed
             // write's callback. This ensures that if the callback calls
             // Grid.write_block() it doesn't preempt the queue.
             if (grid.write_queue.pop()) |queued_write| {
-                const initial_iops_available = grid.write_iops.available();
-                assert(initial_iops_available > 0);
-                grid.start_write(queued_write);
-                assert(grid.write_iops.available() == initial_iops_available - 1);
+                grid.write_block_with(iop, queued_write);
```

Comment:
If this line begins another write on the same address, will the callback below hit an assertion?

Reply:
This is checked by `grid.assert_not_writing(address, block)` in `Grid.write_block()`, so a Write on the same address shouldn't get a chance to push to `write_queue`.